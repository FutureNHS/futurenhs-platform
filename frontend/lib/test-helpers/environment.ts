export const withEnvVars = (vars: { [name: string]: string }) => {
  beforeAll(() => {
    Object.assign(process.env, vars);
  });

  afterAll(() => {
    for (const key of Object.keys(vars)) {
      delete process.env[key];
    }
  });
};
