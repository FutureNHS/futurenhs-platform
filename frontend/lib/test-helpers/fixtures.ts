export const kratosRequestBase = {
  active: "password",
  expiresAt: new Date(),
  id: "598a45d2-e107-41ba-885a-c5c39e4a26d5",
  issuedAt: new Date(),
  messages: undefined,
  // I don't think we ever use this.
  requestUrl: "http://localhost:4455/self-service/browser/flows/settings",
};
