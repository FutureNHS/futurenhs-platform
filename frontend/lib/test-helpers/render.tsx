import React from "react";

import { render as defaultRender, RenderResult } from "@testing-library/react";
import { RouterContext } from "next/dist/next-server/lib/router-context";
import { NextRouter } from "next/router";
import { ThemeProvider } from "styled-components";

import theme from "../../lib/fixtures/theme.json";

export * from "@testing-library/react";

// --------------------------------------------------
// Override the default test render with our own, which includes a router and
// theme mock.
//
// You can override the router mock like this:
//
// const { baseElement } = render(<MyComponent />, {
//   router: { pathname: '/my-custom-pathname' },
// });
// --------------------------------------------------
type DefaultParams = Parameters<typeof defaultRender>;
type RenderUI = DefaultParams[0];
type RenderOptions = DefaultParams[1] & { router?: Partial<NextRouter> };

export function render(
  ui: RenderUI,
  { wrapper, router, ...options }: RenderOptions = {}
): RenderResult {
  if (!wrapper) {
    const Wrapper = ({ children }: { children?: React.ReactNode }) => (
      <RouterContext.Provider value={{ ...mockRouter, ...router }}>
        <ThemeProvider theme={theme}>{children}</ThemeProvider>
      </RouterContext.Provider>
    );
    Wrapper.displayName = "Wrapper";
    wrapper = Wrapper;
  }

  return defaultRender(ui, { wrapper, ...options });
}

const mockRouter: NextRouter = {
  basePath: "",
  pathname: "/",
  route: "/",
  asPath: "/",
  query: {},
  push: jest.fn(),
  replace: jest.fn(),
  reload: jest.fn(),
  back: jest.fn(),
  prefetch: jest.fn(),
  beforePopState: jest.fn(),
  events: {
    on: jest.fn(),
    off: jest.fn(),
    emit: jest.fn(),
  },
  isFallback: false,
};
