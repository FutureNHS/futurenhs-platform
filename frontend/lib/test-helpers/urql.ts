import { DocumentNode } from "graphql";
import { makeResult, Client, Operation, createClient } from "urql";
import { fromValue } from "wonka";

import { Query } from "../generated/graphql";

export type MockUrqlClient = (
  queries: Array<[DocumentNode, Partial<Query>]>
) => Client;

export const mockUrqlClient: MockUrqlClient = (queries) => {
  const client = createClient({ url: "http://localhost", exchanges: [] });
  return Object.assign(client, {
    executeQuery: jest.fn(({ query }) => {
      const operation: Operation = {
        context: {} as any,
        key: 1,
        operationName: "query",
        query,
      };
      const data = queries.find((q) => q[0] === query);
      return fromValue(
        data
          ? makeResult(operation, { data: data[1] })
          : makeResult(operation, {
              error: "query not found",
            })
      );
    }),
  });
};
