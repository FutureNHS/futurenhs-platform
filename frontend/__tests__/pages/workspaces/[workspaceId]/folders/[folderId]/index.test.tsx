import React from "react";

import { ThemeProvider } from "styled-components";

import theme from "../../../../../../lib/fixtures/theme.json";
import {
  FoldersByWorkspaceDocument,
  GetFolderByIdDocument,
  GetWorkspaceByIdDocument,
} from "../../../../../../lib/generated/graphql";
import { render } from "../../../../../../lib/test-helpers/render";
import { mockUrqlClient } from "../../../../../../lib/test-helpers/urql";
import FolderHomepage from "../../../../../../pages/workspaces/[workspaceId]/folders/[folderId]";

describe(FolderHomepage, () => {
  const client = mockUrqlClient([
    [
      GetWorkspaceByIdDocument,
      {
        workspace: {
          id: "1",
          title: "hospital",
          description: "hospital",
        },
      },
    ],
    [
      GetFolderByIdDocument,
      {
        folder: {
          id: "f1",
          title: "folder 1",
          description: "first folder",
          workspace: "1",
        },
      },
    ],
    [
      FoldersByWorkspaceDocument,
      {
        foldersByWorkspace: [
          {
            id: "f1",
            title: "folder 1",
            description: "first folder",
            workspace: "1",
          },
        ],
      },
    ],
  ]);

  test("renders matching snapshot", () => {
    const container = render(
      <ThemeProvider theme={theme}>
        <FolderHomepage urqlClient={client} />
      </ThemeProvider>,
      {
        router: {
          query: { workspaceId: "1", folderId: "f1" },
        },
      }
    );
    expect(container.asFragment()).toMatchSnapshot();
  });
});
