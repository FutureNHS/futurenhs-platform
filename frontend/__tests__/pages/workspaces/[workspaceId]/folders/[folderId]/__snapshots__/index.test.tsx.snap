// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withUrql renders matching snapshot 1`] = `
<DocumentFragment>
  <p>
    Loading...
  </p>
</DocumentFragment>
`;
